server/src/index.mjs

```diff
@@ -41,13 +41,28 @@ const server = createServer();
 server.on('upgrade', function upgrade(request, socket, head) {
   const { pathname } = parse(request.url);
   let userId = pathname.substring(1);
+
   if ( ! userId ) {
     logger.info('Using default user');
     userId = config.app.defaultUserId;
   } else if ( ! userManagement.isKnownUser(userId) ) {
     logger.warn(`WARNING: Unknown userId: ${userId}; Using default user`);
     userId = config.app.defaultUserId;
   }
+  
+  if( commandLine.proxy ) {
+    process.env.proxyServerIP = commandLine.proxy
+    logger.info('Send proxy request to websocket server: ' + process.env.proxyServerIP);
+    process.env.proxy = true
+    // Get token from connection parameter or from env
+    const token = proxyManagement.getToken(request)
+    if( token.stdout ) {
```

Comment:
This seems like a weird protocol (naming a field/member "stdout").

Reply:
You can return an object with two fields if you want to return either a token or an error. E.g., "return { token: null, error: xxx }" or "return { token: xxx, error: null }" and then use these fields here.